detection_rules/integrations.py

```diff
@@ -63,32 +62,25 @@ def build_integrations_manifest(overwrite: bool, rule_integrations: list) -> Non
 
 
 def find_least_compatible_version(package: str, integration: str,
-                                  current_stack_version: str, packages_manifest: dict) -> Union[str, None]:
+                                  current_stack_version: str, packages_manifest: dict) -> str:
     """Finds least compatible version for specified integration based on stack version supplied."""
-    integration_manifests = {k: v for k, v in sorted(packages_manifest[package].items(), key=Version)}
+    integration_manifests = {k: v for k, v in sorted(packages_manifest[package].items(),
```

Comment:
even though dicts started preserving order in py3, I still think it is good practice to not completely rely on it. That is why my suggestion was on the loop, to guarantee that it stay ordered. Otherwise you could consider using an OrderedDict if keeping it here

Reply:
Added `OrderedDict` to iteration object after sorting again by `Version(str(x[0]))` (x[0] is major number in semantic version). Checked that the sorted dictionary is in ascending order as expected to find least compatible.